ui-core/src/main/java/com/adyen/checkout/components/ui/PaymentInProgressView.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 23/9/2022.
+ */
+
+package com.adyen.checkout.components.ui
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.adyen.checkout.components.base.ActionDelegate
+import com.adyen.checkout.components.base.ComponentDelegate
+import com.adyen.checkout.components.ui.databinding.ViewPaymentInProgressBinding
+import com.adyen.checkout.core.exception.CancellationException
+import kotlinx.coroutines.CoroutineScope
+
+class PaymentInProgressView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) :
+    ConstraintLayout(
+        context,
+        attrs,
+        defStyleAttr
+    ),
+    ComponentViewNew {
+
+    private val binding = ViewPaymentInProgressBinding.inflate(LayoutInflater.from(context), this)
+
+    init {
+        layoutParams = LayoutParams(MATCH_PARENT, WRAP_CONTENT)
+    }
+
+    override fun initView(delegate: ComponentDelegate, coroutineScope: CoroutineScope, localizedContext: Context) {
+        if (delegate !is ActionDelegate<*>) throw IllegalStateException("Unsupported delegate type")
```

Comment:
What if we make an interface just for the error throwing delegates? We might use this in other places, like payment components and the only thing we need the `delegate` for is to throw the error, WDYT?